.github/workflows/chaos.yaml

```diff
@@ -0,0 +1,192 @@
+      - name: Print Go version
+        run: go version
+
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - name: Check out code
+        uses: actions/checkout@v2
+
+      - name: Cache Lint Tools
+        id: cache-lint-tools
+        uses: actions/cache@v2
+        with:
+          path: tools/bin
+          key: ${{ runner.os }}-cdc-lint-tools-${{ hashFiles('tools/check/go.sum') }}
+
+      - name: Set up K3s cluster
+        run: |
+          curl -sfL https://get.k3s.io | INSTALL_K3S_VERSION=v1.18.9+k3s1 sh -s - \
+            --write-kubeconfig-mode=644 \
+            "${k3s_disable_command:---disable}" metrics-server \
+            "${k3s_disable_command:---disable}" traefik \
+            --flannel-backend=none \
+            --docker
+        shell: bash
+
+    # this may be failed sometimes, and I want to exit the workflow directly if failed,
+    # but GitHub Actions doesnt' support early-exit yet, see https://github.com/actions/runner/issues/662.
+    # so, simply wait for a long time.
+      - name: Wait for coredns
+        run: |
+          kubectl rollout status --watch --timeout 600s deployment/coredns -n kube-system
+        shell: bash
+        env:
+          KUBECONFIG: /etc/rancher/k3s/k3s.yaml
+
+      - name: Export KUBECONFIG environment variable
+        run: |
+          echo 'KUBECONFIG=/etc/rancher/k3s/k3s.yaml' >> $GITHUB_ENV
+        shell: bash
+
+      - name: Print cluster information
+        run: |
+          kubectl config view
+          kubectl cluster-info
+          kubectl get nodes
+          kubectl get pods -n kube-system
+          kubectl get sc
+          kubectl version
+
+      - name: Install tidb-operator
+        run: |
+          echo "install helm 3"
+          curl -fsSL -o get_helm.sh https://raw.githubusercontent.com/helm/helm/master/scripts/get-helm-3
+          chmod 700 get_helm.sh
+          ./get_helm.sh
+
+          echo "install tidb-operator CRDs"
+          kubectl apply -f https://raw.githubusercontent.com/pingcap/tidb-operator/v1.1.12/manifests/crd.yaml
+          kubectl get crd
+
+          echo "install tidb-operator into namespace playground"
+          helm repo add pingcap https://charts.pingcap.org/
+
+          kubectl create namespace playground
+          helm install --namespace playground tidb-operator pingcap/tidb-operator --version v1.1.12
+
+      - name: Wait for tidb-operator Ready
+        run: |
+          kubectl wait --for=condition=Ready pod -l app.kubernetes.io/instance=tidb-operator -n playground --timeout=300s || true
+
+          kubectl get pods --namespace playground -l app.kubernetes.io/instance=tidb-operator
+
+      - name: Deploy Upstream / Downstream TiDB Cluster into playground
+        run: |
+          kubectl apply -f $GITHUB_WORKSPACE/chaos/manifests/upstream.yaml -n playground
+          kubectl apply -f $GITHUB_WORKSPACE/chaos/manifests/downstream.yaml -n playground
+
+          sleep 100
```

Comment:
no, we have to sleep at least 100 seconds. For tidb-operator, it will start the tidb pod at last, before tidb pod exist, `kubectl wait` will fail because it does not exist yet.

Reply:
Maybe we can use [`wait-for-it`](http://manpages.ubuntu.com/manpages/bionic/man8/wait-for-it.8.html) to wait for TiDB's listen port to be ready.

Reply:
`wait-for-it upstream-tidb.playground.svc:4000  --timeout=100` 

consider this, wait-for-it cannot visit k8s service URL, this will fail. so `sleep 100` may be the final solution.